src/DynamoCore/Nodes/Attributes.cs

```diff
@@ -208,4 +208,173 @@ public NodeObsoleteAttribute(string descriptionResourceID, Type resourceType)
             }
         }
     }
+
+    /// <summary>
+    /// PortAttribute is base class for all attibutes used for ports.
+    /// PortAttribute can process resource file (.resx) and load strings from this file.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class)]
+    public abstract class PortAttribute : Attribute
```

Comment:
Instead of a base class, let's just make a small `internal` utility class that can read data from a resources type as done in the constructor. This gives us a bit more flexibility about the way this functionality evolves.